File: week2/day2/draw-diagonal.py
# Write a program that reads a number from the standard input, then draws a
# square like this:
#
#
# %%%%%
# %%  %
# % % %
# %  %%
# %   %
# %%%%%
#
# The square should have as many lines as the number was

x = int(input("Please enter a number for the square: "))
a = 0

print((x-1) * "%")

while a < (x - 2):
    a += 1
    print("%" + (a-1) * " " + ((((2 * x) - 3) // (a + x)) * "%") + ((x - 3) - a) * " " + "%")

print((x-1) * "%")

File: week8/day2/draw-diagonal.js
'use strict';

var lineCount = 6;


// Write a program that draws a
// square like this:
//
//
// %%%%%
// %%  %
// % % %
// %  %%
// %   %
// %%%%%
//
// The square should have as many lines as lineCount is

var block = '%'
var space = ' '

console.log(block.repeat((lineCount-1)));
for (var i = 0; i < (lineCount - 3); i++) {
    console.log(block + space.repeat(i) + block + space.repeat((lineCount - 4 - i)) + block);
}
console.log(block + space.repeat(lineCount - 3) + block);
console.log(block.repeat((lineCount-1)));
